pkg/core/handler/api.go

```diff
@@ -79,17 +86,27 @@ func APIHandlerToHandler(apiHandler APIHandler, txContext db.TxContext) http.Han
 	})
 }
 
+type HandledError struct {
+	Error error
+}
+
 func WriteResponse(rw http.ResponseWriter, response APIResponse) {
 	httpStatus := http.StatusOK
 	encoder := json.NewEncoder(rw)
+	err := skyerr.AsAPIError(response.Error)
 
-	if response.Error != nil {
-		httpStatus = response.Error.Code
+	if err != nil {
+		httpStatus = err.Code
 	}
 
 	rw.Header().Set("Content-Type", "application/json")
 	rw.WriteHeader(httpStatus)
 	encoder.Encode(response)
+
+	if err != nil && err.Code >= 500 && err.Code < 600 {
+		// delegate logging to panic recovery
```

Comment:
Can we document the struct `HandledError` instead? For example, mention it is meant to be used as sentinel value.